backend/httpclient/config/response_limit_middleware.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+)
+
+// ResponseLimitMiddlewareName is the middleware name used by ResponseLimitMiddleware.
+const ResponseLimitMiddlewareName = "response-limit"
+
+func ResponseLimitMiddleware(limit int64) httpclient.Middleware {
+	return httpclient.NamedMiddlewareFunc(ResponseLimitMiddlewareName, func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+		return httpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			res, err := next.RoundTrip(req)
+			if err != nil {
+				return nil, err
+			}
+
+			reqLimit := limit
+			if reqLimit <= 0 {
+				reqLimit = backend.GrafanaConfigFromContext(req.Context()).ResponseLimit()
+			}
```

Comment:
this is the new code for the middleware (comparing to the version in core).

Reply:
I guess a new package is needed due to import cycles putting this in the httpclient package? This might come and bite us back later, so wonder if we really should have this here rather than require the consumer to always provide the config for now. The old code is not executing the middleware/disables it if the limit is <= 0 so this changes the behavior. 

Eventually we might need to figure out how to provide a better/an easier way creating HTTP clients with default middlewares for external/decoupled plugins and that needs to take multi tenancy into account which might suggest HTTP client provider is no longer a good fit?